frontend/src/metabase/querying/notebook/components/NotebookDataPicker/NotebookDataPicker.tsx

```diff
@@ -193,30 +195,69 @@ type LegacyDataPickerProps = {
   stageIndex: number;
   table: Lib.TableMetadata | Lib.CardMetadata | undefined;
   placeholder: string;
+  canChangeDatabase: boolean;
   hasMetrics: boolean;
   isDisabled: boolean;
   onChange: (tableId: TableId) => void;
 };
 
-function LegacyDataPicker({
+function EmbeddingDataPicker({
   query,
   stageIndex,
   table,
   placeholder,
+  canChangeDatabase,
   hasMetrics,
   isDisabled,
   onChange,
 }: LegacyDataPickerProps) {
+  const { data: dataSourceCountData, isLoading: isDataSourceCountLoading } =
+    useSearchQuery({
+      models: ["dataset", "table"],
+      limit: 0,
+    });
+
   const databaseId = Lib.databaseID(query);
   const tableInfo =
     table != null ? Lib.displayInfo(query, stageIndex, table) : undefined;
+  /**
+   * We try to blur the line between models and tables for embedding users.
+   * this property will change the way icons are displayed in the data picker trigger,
+   * so we need to remove it. Treating it as a table.
+   */
+  const tableInfoWithTableIcon = tableInfo && {
```

Comment:
If you don't need an icon, please add `withIcon` to `DataPickerTarget` and set it to true where needed. When it's not set, render a table icon. Please do not modify objects coming from CLJS as they are memoized

Reply:
Ahh, that's great to point out that we shouldn't modify objects from CLJS 🙏 TIL.

Fixed by c78ad24eccd41f1c92364a6d4c56962ce2241529